doc/config.rst

```diff
@@ -171,6 +171,21 @@ General configuration
 
    .. versionadded:: 1.0
 
+.. confval:: section_titles_as_targets
+
+   The default is ``False``.
```

Comment:
Starting with the default value is very odd.  I'd rephrase the whole section here… first, say what this setting is for, then what the values will do, and close with the default value.